### PR TITLE
Add new caching options

### PR DESCRIPTION
**Pull request by enumag**

### CODE REVIEW

**Comment by enumag** [comment]
Part of https://github.com/Kdyby/Annotations/issues/7, not ready to merge yet.

File: src/Kdyby/DoctrineCache/DI/Helpers.php [hunk 1]
```diff
@@ -60,6 +62,10 @@ public static function processCache(Nette\DI\CompilerExtension $extension, $cach
 			$cache->arguments[2] = $debug !== NULL ? $debug : $builder->parameters['debugMode'];
 		}
 
+		if ($impl === 'filesystem') {
+			$cache->arguments[] = $builder->parameters['tempDir'] . '/Doctrine.' . ucfirst($suffix);
```
Comment (fprochazka) [inline src/Kdyby/DoctrineCache/DI/Helpers.php]: afaik `tempDir` does not contain `cache/`

**Comment by fprochazka** [comment]
Looks good. I'd merge this and tag new version, whad do you think?

**Comment by enumag** [comment]
@fprochazka I'd like to test if all the PRs together really do what I've intended. Give me a day or so. ;-)

**Comment by fprochazka** [comment]
@enumag works for me, I wanned to go to sleep anyway :)

**Comment by enumag** [comment]
I'm trying the filesystem cache now and it creates really deep directory structures. For example:

```
/temp/cache/Doctrine.Validator/41/d3/9f/97/c4/89/2f/9d/cf/a3/1f/01/71/1c/fc/98/6b/d2/d5/2f/1f/79/a9/46/68/a3/9f/38/8f/b4/ed/b7/DoctrineNamespaceCacheKey[Kdyby_kdyby-validator-cache-validator_c4bbc638].doctrinecache.data
```

I really hate this behaviour. :-/

It's because of [this line](https://github.com/doctrine/cache/blob/master/lib/Doctrine/Common/Cache/FileCache.php#L129).

What is your opinion about this behaviour? I could extend the FilesystemCache and replace the method with something different.

**Comment by enumag** [comment]
Aside from this the PRs indeed do what I've intended. :-)

**Comment by enumag** [comment]
Maybe we should just wait for doctrine/cache v1.5 (see https://github.com/doctrine/cache/pull/94 for details).

**Comment by enumag** [comment]
@fprochazka What's your opinion?

**Comment by fprochazka** [comment]
@enumag I extend the FilesystemCache and override the method with custom impl.

**Comment by enumag** [comment]
@fprochazka I'd like this merged even without the fix for FilesystemCache. It won't be default anywhere unless we make it that way - which we obviously won't until the issue is solved. It can still work as an optional caching option in the meantime.

**Comment by fprochazka** [comment]
Agreed.

**Pull request merged**